### ISSUE TITLE
Dimension Mismatch 

### ISSUE THREAD

**Issue by Croftc** [dscotthunter/Fantasy-Hockey-IP-Code#5]
I am successfully able to run all provided formulations on the provided test data, but running on actual data, formulations 3 and 4 don't work - specifically the defensemen stacking constraint. The error I receive states:

glp_intopt: optimal basis to initial LP relaxation not provided
WARNING: Not solved to optimality, status: Infeasible
WARNING: Infeasibility ray (Farkas proof) not available
ERROR: LoadError: DimensionMismatch("mismatch in dimension 1 (expected 1 got 217)")

**Comment by jcameronmetz** [comment]
Did you ever get this working I'm having the same issue?  The (Array{Int64,0})[]  was a big fix

**Comment by jcameronmetz** [comment]
This was also given : DimensionMismatch("mismatch in dimension 1 (expected 1 got 360)")

**Comment by trbradle44** [comment]
make sure you have the teams/opponents laid out correctly.
for example:
     team A with opp team B, & team B with opp team A; NOT,
     team A with opp team B, & team B with opp team C

hope that helps.

**Comment by shane-rand** [comment]
Any luck on this? I'm running into the same error and I verified teams and opponents matching up.

**Comment by jcameronmetz** [comment]
Yep- I'll respond this weekend at some point

**Comment by shane-rand** [comment]
@jcameronmetz Can you let me know how you fixed this?

**Comment by jcameronmetz** [comment]
@constraint needs to be changed

@variable needs changed

Those are the things that helped work for me.

**Comment by jcameronmetz** [comment]
# Variable for skaters in lineup.
    @defVar(m, skaters_lineup[i=1:num_skaters], Bin)   -> change this to @variable

    # Variable for goalie in lineup.
    @defVar(m, goalies_lineup[i=1:num_goalies], Bin)


    # One goalie constraint
    @addConstraint(m, sum{goalies_lineup[i], i=1:num_goalies} == 1)  change this to @constraint

**Comment by shane-rand** [comment]
@jcameronmetz Hmm, unfortunately I already had those fixes in my code. Still experiencing the issue. I believe since the model is not being solved, its throwing the error on the next lineup creation.

The example data works perfectly, leading me to believe I've set up my data wrong? If you could send over your last data file that worked (both skaters.csv and goalies.csv) that would be appreciated.

**Comment by shane-rand** [comment]
Still working on this....

I've found that its the 'at least one full line' and 'defenseman on powerplay one' constraints that are the problem. Currently trying to decipher those constraints...

**Comment by jcameronmetz** [comment]
Yeah - making your lines correct isnkey - try running a report against your
lines to make sure you have complete lines

**Comment by shane-rand** [comment]
Finally figured it out.

https://github.com/dscotthunter/Fantasy-Hockey-IP-Code/blob/673b0a5119ed746a8dc2347206d138c84407def1/code_for_Github.jl#L736-L799

These lines are using double quotes around the line numbers for comparison. I removed the quotes and compared as a number and everything worked swimmingly.

**Comment by awiden91** [comment]
@shane-rand Hey, could you please send over the full code that you used to get it running? I am having trouble updating it and keep running into a LoadError: UndefVarError: i not defined error. Would you be able to help me with this. Thank you and I appreciate your time!

**Comment by mhbw** [comment]
@awiden91 also hitting the "I not defined" issue; did you ever solve it?
